test/junit/scala/collection/immutable/LazyListTest.scala

```diff
@@ -349,13 +396,13 @@ class LazyListTest {
 
     // Check forced cyclic and non-cyclic streams
     assertEquals("LazyList(-2, -1)", pre(2).force.toString)
-    assertEquals("LazyList(0, 1, <cycle>)", cyc(2).force.toString)
-    assertEquals("LazyList(-2, -1, 0, 1, <cycle>)", precyc(2,2).force.toString)
+    assertEquals("LazyList(0, 1, 0, <cycle>)", cyc(2).force.toString)
```

Comment:
it occurred to me that we could almost certainly prevent the duplicate element by buffering the last element in a `var` and appending it "late" (and skipping that append if we detect a cycle)

Reply:
Right, thank you for the idea. I cleaned up `addStringNoForce` and implemented your suggestion.